browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -302,6 +345,38 @@ void SpeedreaderTabHelper::DidStopLoading() {
   }
 }
 
+void SpeedreaderTabHelper::DOMContentLoaded(
+    content::RenderFrameHost* render_frame_host) {
+  if (!PageWantsDistill(distill_state_))
+    return;
+
+  constexpr const char16_t kAddShowOriginalPageLink[] =
+      uR"js(
+    (function() {
+      // element id is hardcoded in extractor.rs
+      const link = document.
+        getElementById('c93e2206-2f31-4ddc-9828-2bb8e8ed940e');
+      if (!link)
+        return;
+      link.text = '$1';
+      link.addEventListener('click', (e) => {
+        window.speedreader.showOriginalPage();
+      })
+    })();
+  )js";
+
+  const auto link_text = brave_l10n::GetLocalizedResourceUTF16String(
+      IDR_SPEEDREADER_SHOW_ORIGINAL_PAGE_LINK);
+  // Make sure that the link text doesn't contain js injection
+  CHECK_EQ(std::u16string::npos, link_text.find(u'\''));
```

Comment:
maybe `\u0027` could still be used to inject `'`

could just add another CHECK that it doesn't contain `\`

Reply:
Done